minpy/nn/solver.py

```diff
@@ -178,14 +193,27 @@ def loss_func(*params):
 
         self.loss_history.append(loss.asnumpy())
 
+        if self.visualize:
+            grad_norm = 0
+
         # Perform a parameter update
         for p, w in self.model.params.items():
             dw = grads[p]
+            if self.visualize:
+                norm = dw ** 2
+                while not isinstance(norm, minpy.array.Number):
+                    norm = sum(norm)
+                grad_norm += norm
             config = self.optim_configs[p]
             next_w, next_config = self.update_rule(w, dw, config)
             self.model.params[p] = next_w
             self.optim_configs[p] = next_config
 
+        if self.visualize:
+            grad_norm_summary = summary_ops.scalar_summary('squared L2-norm of the gradient',
+                                                           grad_norm)
+            self.train_writer.add_summary(grad_norm_summary, iteration)
+
```

Comment:
This part of logic could be simply put in the visualizer object I mentioned above.